### PR TITLE
Settings API

### PR DESCRIPTION
**Pull request by gearama**
KV Settings api 
* API 
* Test
* Sample
* Docs
* GetRandomBytes test now has recording, since i have an HSM available. 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+#if defined(TESTING_BUILD)
+
+  namespace Test {
+    class SettingsClientTest;
+  }
+#endif
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:26]: This should be isolated to a _test.hpp header.  Why does it need to be in the public header?

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  class SettingsClient
+#if !defined(TESTING_BUILD)
+      final
+#endif
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:31]: If we need to extend to test, why wouldn't a customer?

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:31]: removed the #if , not needed anymore.

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp (hunk 1 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:26]: removed, not needed anymore

**Review by ahsonkhan — COMMENTED** [review]
Is this PR ready for review?

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/dll_import_export.hpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * @brief Azure SDK abstractions.
+ *
+ */
+namespace Azure {
+
+/**
+ * @brief Azure Template SDK abstractions.
+ *
+ */
+namespace Template {
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/dll_import_export.hpp:52]: Update this file, as needed, still says `Template`.

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/dll_import_export.hpp]: done

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  endif()
+endif()
+
+set(
+  AZURE_SECURITY_KEYVAULT_ADMINISTRATION_HEADER
+    inc/azure/keyvault/administration/dll_import_export.hpp
+    inc/azure/keyvault/administration/rtti.hpp
+    inc/azure/keyvault/administration/settings_client.hpp
+    inc/azure/keyvault/administration/settings_client_options.hpp
+    inc/azure/keyvault/administration/rest_client.hpp
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:54]: Alphabetize the includes.

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    inc/azure/keyvault/administration.hpp
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:55]: This should be the first one in the list

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    src/private/keyvault_settings_common_request.hpp
+    src/private/administration_constants.hpp
+)
+
+set(
+  AZURE_SECURITY_KEYVAULT_ADMINISTRATION_SOURCE
+    src/settings_client.cpp
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:62]: Alphabetize

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+generate_documentation(azure-security-keyvault-administration ${AZ_LIBRARY_VERSION})
+if(BUILD_TESTING)
+
+  if (NOT AZ_ALL_LIBRARIES OR FETCH_SOURCE_DEPS)
+    include(AddGoogleTest)
+    enable_testing ()
+  endif()
+  
+  add_subdirectory(test/ut)
+ # add_subdirectory(test/test-app)
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:100]: Remove the commented line
```suggestion
```

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+endif()
+
+if (BUILD_PERFORMANCE_TESTS)
+  #add_subdirectory(test/perf)
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:104]: Will we add tests?  If so add a work item to track, if not remove the #def

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+if(BUILD_SAMPLES)
+  add_subdirectory(test/samples)
+endif()
+
+az_vcpkg_export(
+    azure-security-keyvault-administration
+    ration
+    SECURITY_KEYVAULT_ADMINISTRATION
+    "azure/keyvault/administration/dll_import_export.hpp"
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:115]: Why is this line here when its part of the includes in azure-security-keyvault-administration

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+  )
+
+az_rtti_setup(
+  azure-security-keyvault-administration
+  SECURITY_KEYVAULT_ADMINISTRATION
+  "azure/keyvault/administration/rtti.hpp"
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:121]: This is an include in the azure-security-keyvault-administration.  Line is unnecessary

File: sdk/keyvault/azure-security-keyvault-administration/NOTICE.txt [hunk 11]
```diff
@@ -0,0 +1,32 @@
+azure-template
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/NOTICE.txt:1]:
```suggestion
azure-security-keyvault-administration
```

File: sdk/keyvault/azure-security-keyvault-administration/README.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
+# Azure Security Keyvault Administration Package client library for C++
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/README.md:1]: Why is `Package` in the name?  Unless that is standard in all languages I would prefer
`Azure Security KeyVault Administration client library for C++`

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration.hpp [hunk 13]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief Includes all public headers from Azure Template SDK library.
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration.hpp:5]:
```suggestion
 * @brief Includes all public headers from Azure Security KeyVault Administration SDK library.
```

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 14]
```diff
@@ -0,0 +1,97 @@
+
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:1]:
```suggestion
```

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:15]:
```suggestion
```

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 16]
```diff
@@ -0,0 +1,97 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:3]:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// SPDX-License-Identifier: MIT
```

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+  class SettingTypeEnum final {
+  public:
+    /**
+     * @brief Default constructor
+     */
+    SettingTypeEnum() = default;
+
+    /**
+     * @brief Constructor
+     * @param value string value
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:37]: I suspect the value is something more than just a string.  Are there values that are acceptable or not acceptable.

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    const std::string& ToString() const { return m_value; }
+    AZURE_SECURITY_KEYVAULT_ADMINISTRATION_DLLEXPORT const static SettingTypeEnum Boolean;
+
+  private:
+    std::string m_value;
+  };
+
+  /**
+   * @brief Update Security Options
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:56]: The brief is `Update Security Options` yet the struct is `UpdateSettingOptions` which one is the right one?  They should match

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+  struct UpdateSettingOptions final
+  {
+    /**
+     * The value of the pool setting.
+     */
+    std::string Value;
+  };
+
+  /**
+   * @brief Setting Model
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:67]: We shouldn't call it a Model.  Its the "Setting" which is a struct/type/model/...  Lets standardize on simply referring to it as the type name.  I suspect this is a codegen bug

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * The type specifier of the value.
+     */
+    Azure::Nullable<SettingTypeEnum> Type;
+  };
+
+  /**
+   * @brief The settings list result.
+   */
+  struct SettingsListResult final
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp:88]: Does the postfix `Result` make sense or should it be `SettingsList`?  This is another item that is probably codegen and we need to make sure it follows a consistent pattern and is logical to the customer

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+#include <azure/core/context.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <azure/core/response.hpp>
+#include <azure/keyvault/administration/rest_client.hpp>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  class SettingsClient final {
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:21]: Missing doc comment on the class

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Defines the Key Vault Administration client.
+ *
+ */
+
+#pragma once
+#include "settings_client_options.hpp"
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:11]: Why isn't this
`<azure/keyvault/administration/settings_client_options.hpp>`

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+#include <azure/core/response.hpp>
+#include <azure/keyvault/administration/rest_client.hpp>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  class SettingsClient final {
+
+  private:
+    // Using a shared pipeline for a client to share it with LRO (like delete key)
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:24]: This comment should be on the line before the shared_ptr for the pipeline

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+     *
+     * @param settingsClient An existing key vault settings client.
+     */
+    explicit SettingsClient(SettingsClient const& settingsClient) = default;
+
+  public:
+    /**
+     * @brief Update a setting
+     *
+     * @param settingName Name of the setting to update
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp:59]: Be consistent with `.` punctuation.  Some params have them, some don't.  Our standard is to have them.

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client_options.hpp [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+ * @file
+ * @brief Defines the supported options to create a Key Vault Administration client.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/internal/client_options.hpp>
+
+#include "azure/keyvault/administration/dll_import_export.hpp"
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client_options.hpp:14]: Why is this not
`<azure/keyvault/administration/dll_import_export.hpp>`

File: sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client_options.hpp [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+
+  /**
+   * @brief Define the options to create an SDK Administration client.
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client_options.hpp:21]:
```suggestion
   * @brief Define the options to create a KeyVault Administration client.
```

File: sdk/keyvault/azure-security-keyvault-administration/samples/README.md [hunk 27]
```diff
@@ -0,0 +1,14 @@
...
+products:
+  - azure
+  - azure-key-vault
+name: Azure Security KeyVault Administration samples for C++
+description: Samples for the azure-security-keyVault-administration client library.
+---
+
+# Azure.Security.KeyVault.Secrets Samples
+
+- Get Settings, Get Setting , Update setting value
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/samples/README.md:14]: Nit: We return a List for GetSettings.  Perhaps we should standardize on `List`Settings when returning multiple items.  No need to change now, but do have the conversation in the arch review

File: sdk/keyvault/azure-security-keyvault-administration/samples/sample1_administration.md [hunk 28]
```diff
@@ -0,0 +1,63 @@
+# Getting, updating, settings
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/samples/sample1_administration.md:1]: The name of the file is `sample1_adminstration.md` why the `1`?  Can it be `sample_adminstration.md`?

File: sdk/keyvault/azure-security-keyvault-administration/src/private/administration_constants.hpp [hunk 29]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Centralize the string constants used by Key Vault Certificates Client.
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/private/administration_constants.hpp:6]:
```suggestion
 * @brief Centralize the string constants used by Key Vault Administration Client.
```

File: sdk/keyvault/azure-security-keyvault-administration/src/private/keyvault_settings_common_request.hpp [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+#include <azure/core/internal/http/pipeline.hpp>
+#include <azure/core/internal/json/json.hpp>
+#include <azure/core/internal/json/json_serializable.hpp>
+#include <azure/core/response.hpp>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace _detail {
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/private/keyvault_settings_common_request.hpp:23]: Why is this at the `KeyVault::_detail` level and not at the `KeyVault::Administration::_detail` level?

File: sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp [hunk 31]
```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/keyvault/shared/keyvault_shared.hpp"
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp:4]:
```suggestion
#include <azure/keyvault/shared/keyvault_shared.hpp>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp [hunk 32]
```diff
@@ -0,0 +1,154 @@
...
+using namespace Azure::Security::KeyVault::Administration::_detail;
+using namespace Azure::Security::KeyVault;
+using namespace Azure::Security::KeyVault::_detail;
+
+using namespace Azure;
+using namespace Azure::Core;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
+using namespace Azure::Core::Http::_internal;
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp:26]: Alphabetize.  Are all of these necessary?

File: sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp [hunk 33]
```diff
@@ -0,0 +1,154 @@
...
+
+using namespace Azure::Core::Json::_internal;
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp:28]: Why is this one off by itself?

File: sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp [hunk 34]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+Setting SettingsClient::ParseSetting(std::vector<uint8_t> const& responseBody) const
+{
+  Setting response;
+  {
+    auto jsonRoot
+        = Azure::Core::Json::_internal::json::parse(responseBody.begin(), responseBody.end());
+    response.Name = jsonRoot[NameField].get<std::string>();
+    response.Value = jsonRoot[ValueField].get<std::string>();
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp:147]: Do we get `null` or `""` or an `access violation` if the field isn't present?

File: sdk/keyvault/azure-security-keyvault-administration/vcpkg/vcpkg.json [hunk 35]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+{
+  "name": "azure-security-keyvault-administration-cpp",
+  "version-semver": "@AZ_LIBRARY_VERSION@",
+  "description": [
+    "Microsoft Azure Key Vault Administratio SDK for C++",
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-administration/vcpkg/vcpkg.json:8]:
```suggestion
    "Microsoft Azure Key Vault Administration SDK for C++",
```

**Review by RickWinter — APPROVED** [review]
Please add more tests around negative cases (invalid payloads, error responses, etc...)

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 4 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:54]: done

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 5 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:55]: done

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 6 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:62]: done

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 7 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:100]: done

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 8 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:104]: we will

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 9 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:115]: cmake erros out withou tit

File: sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt (hunk 10 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt:121]: cmake erros out without it

File: sdk/keyvault/azure-security-keyvault-administration/samples/sample1_administration.md (hunk 28 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/samples/sample1_administration.md:1]: it Sample One : Administration , when we add RBAC and others it will be Sample 2 : RBAC this is episode 1 of the samples

File: sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp (hunk 34 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp:147]: you cannot have a field without name and value in json

**Comment by gearama** [comment]
need some part of this PR for the OKP PR . I will merge it and we can make changes to names if needed in a subsequent PR
